src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -308,14 +320,17 @@ public void OnResponseHeader(ReadOnlySpan<byte> name, ReadOnlySpan<byte> value)
                         // if the header can't be added, we silently drop it.
                         if (_state == StreamState.ExpectingTrailingHeaders)
                         {
-                            _response.TrailingHeaders.TryAddWithoutValidation(descriptor.HeaderType == HttpHeaderType.Request ? descriptor.AsCustomHeader() : descriptor, headerValue);
+                            _trailers ??= new List<KeyValuePair<HeaderDescriptor, string>>();
```

Comment:
Seems like we should be able to deterministically create this in OnResponseHeadersStart, then assert here that it's created

Reply:
Yup. Will change it.